### PR TITLE
Schema for Galleries & Carousel

### LINKED ISSUE #1750: Support Schema for Tiled Gallery and Carousel
Would be great if the Tiled Gallery and Carousel could have schema support.

http://schema.org/ImageGallery
http://schema.org/ImageObject

### PR DESCRIPTION
**Pull request by bhubbard**
This PR adds basic schema support to Tiled Galleries and the Carousel
Modules.

Fixes #1750

### CODE REVIEW

File: modules/carousel/jetpack-carousel.js [hunk 1]
```diff
@@ -225,12 +225,15 @@ jQuery(document).ready(function($) {
 
 			container = $('<div></div>')
 				.addClass('jp-carousel-wrap')
-				.addClass( 'jp-carousel-transitions' );
-
+				.addClass( 'jp-carousel-transitions' )
```
Comment (kraftbj) [inline modules/carousel/jetpack-carousel.js]: Travis failure due to jshint picking up the missing semicolon at this line. Could you add it back?

**Comment by eliorivero** [comment]
Thanks @bhubbard for the PR!
I tested this with Google's [structured data testing tool](https://developers.google.com/structured-data/testing-tool/) and it works fine when you enter the HTML manually. It displays the gallery and its elements perfectly
gallery-schema
It seems the fetch tool doesn't execute the JS that adds some schema elements since it doesn't show it when you use Fetch URL. However, [Google understands JavaScript](https://googlewebmastercentral.blogspot.com.ar/2014/05/understanding-web-pages-better.html) so this won't be an issue (provided that users haven't restricted access to JS files).

The one caveat I've found is that if you select Slideshow as layout
slideshow
the schema data isn't added. It would be good if you could add this to the slideshow shortcode so your implementation is complete. This is the [PHP file](https://github.com/Automattic/jetpack/blob/master/modules/shortcodes/slideshow.php) and the [JS file](https://github.com/Automattic/jetpack/blob/master/modules/shortcodes/js/slideshow-shortcode.js) related.

Great work!

**Comment by bhubbard** [comment]
@eliorivero I updated the two files to support the slideshow layout. When you paste code into the [Google's structured data testing tool](https://developers.google.com/structured-data/testing-tool/) it will validate as a gallery, but not show anymore as the rest of the code is generated with javascript.

**Comment by eliorivero** [comment]
Great work @bhubbard! This is working fine :+1:

**Pull request merged**